From: Eric Dorland <eric@debian.org>
Date: Mon, 11 Aug 2008 02:56:28 -0400
Subject: Set the profile path explicitly

---
 browser/app/application.ini |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/browser/app/application.ini b/browser/app/application.ini
index a5fe3ca..7e5f168 100644
--- a/browser/app/application.ini
+++ b/browser/app/application.ini
@@ -39,6 +39,7 @@
 [App]
 Vendor=Mozilla
 Name=Iceweasel
+Profile=mozilla/firefox
 Version=@APP_VERSION@
 BuildID=@GRE_BUILDID@
 #ifdef MOZ_SOURCE_REPO
