From c98341652b24f3ee10a45c04fc3ac6c1af2db597 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Mon, 14 Jul 2008 10:08:56 +0200
Subject: [PATCH 09/17] Avoid spurious "Run" items in application handlers configuration pane

---
 browser/components/preferences/applications.js |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/browser/components/preferences/applications.js b/browser/components/preferences/applications.js
index 4b9d79f..6b71b3c 100644
--- a/browser/components/preferences/applications.js
+++ b/browser/components/preferences/applications.js
@@ -1362,6 +1362,7 @@ var gApplicationsPane = {
     return aExecutable &&
            aExecutable.exists() &&
            aExecutable.isExecutable() &&
+           aExecutable.isFile() &&
 // XXXben - we need to compare this with the running instance executable
 //          just don't know how to do that via script...
 // XXXmano TBD: can probably add this to nsIShellService
-- 
1.7.0.rc0.69.g2ee80

