From: Mike Hommey <mh@glandium.org>
Date: Wed, 8 Jul 2009 08:02:11 +0200
Subject: browser/components/build/Makefile.in: Link against nspr

---
 browser/components/build/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/browser/components/build/Makefile.in b/browser/components/build/Makefile.in
index f22b550..3a4a51c 100644
--- a/browser/components/build/Makefile.in
+++ b/browser/components/build/Makefile.in
@@ -76,7 +76,7 @@ ifndef WINCE
 REQUIRES += migration
 LOCAL_INCLUDES += -I$(srcdir)/../migration/src
 SHARED_LIBRARY_LIBS += ../migration/src/$(LIB_PREFIX)migration_s.$(LIB_SUFFIX)
-EXTRA_DSO_LDOPTS += $(LIBXUL_DIST)/lib/$(LIB_PREFIX)mozreg_s.$(LIB_SUFFIX)
+EXTRA_DSO_LDOPTS += $(LIBXUL_DIST)/lib/$(LIB_PREFIX)mozreg_s.$(LIB_SUFFIX) $(NSPR_LIBS)
 endif
 
 # This has to come after the above chunk, because mozreg_s has dependencies on
