From: Mike Hommey <mh@glandium.org>
Date: Fri, 2 Apr 2010 21:00:08 +0200
Subject: Additional --with-libxul-sdk changes for 3.6

---
 configure.in |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index 5e85de0..fc1ee10 100644
--- a/configure.in
+++ b/configure.in
@@ -4199,6 +4199,7 @@ AC_SUBST(LIBXUL_SDK)
 
 if test -n "$LIBXUL_SDK"; then
     LIBXUL_DIST="$LIBXUL_SDK"
+    SKIP_LIBRARY_CHECKS=1
 else
     LIBXUL_DIST="$MOZ_BUILD_ROOT/dist"
 fi
@@ -8412,7 +8413,7 @@ if test "$MOZ_X11"; then
 fi # MOZ_X11
 fi # COMPILE_ENVIRONMENT
 
-if test "$USE_FC_FREETYPE"; then
+if test "$USE_FC_FREETYPE" && test -z "$LIBXUL_SDK"; then
     if test "$COMPILE_ENVIRONMENT"; then
     	_SAVE_CPPFLAGS="$CPPFLAGS"
     	CPPFLAGS="$CPPFLAGS $FT2_CFLAGS $XCFLAGS"
