From 4460a95f22fbbbea0982cb878872b1902897b2f2 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Sat, 21 Jun 2008 14:43:40 +0200
Subject: [PATCH 14/15] Force to not use -fshort-wchar

---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index 32eaa11..0f82d69 100644
--- a/configure.in
+++ b/configure.in
@@ -2848,7 +2848,7 @@ AC_CACHE_CHECK(for usable wchar_t (2 bytes, unsigned),
 if test "$ac_cv_have_usable_wchar_v2" = "yes"; then
     AC_DEFINE(HAVE_CPP_2BYTE_WCHAR_T)
     HAVE_CPP_2BYTE_WCHAR_T=1
-else
+elif false; then
 dnl This is really gcc-only
 dnl Do this test using CXX only since some versions of gcc
 dnl 2.95-2.97 have a signed wchar_t in c++ only and some versions
-- 
1.7.0.rc0.69.g2ee80

