From 0275a22057a1104e20238c273ecc0057856435ce Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Wed, 24 Feb 2010 18:38:43 +0100
Subject: [PATCH 10/11] mozilla-config.h was renamed js-confdefs.h in js/src

---
 js/src/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/js/src/Makefile.in b/js/src/Makefile.in
index 0f790ab..d57536e 100644
--- a/js/src/Makefile.in
+++ b/js/src/Makefile.in
@@ -397,7 +397,7 @@ distclean::
 
 DIST_GARBAGE = config.cache config.log config.status \
    config/autoconf.mk \
-   unallmakefiles js-config js-config.h mozilla-config.h
+   unallmakefiles js-config js-config.h js-confdefs.h
 
 distclean::
 	cat unallmakefiles | $(XARGS) rm -f
-- 
1.7.0.rc0.69.g2ee80

