From 97bb53a8ecf00b326ad9d23054477b0659e93081 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Wed, 24 Feb 2010 18:40:00 +0100
Subject: [PATCH 11/11] Remove the js shell from the build directory during cleanup

---
 js/src/shell/Makefile.in |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/js/src/shell/Makefile.in b/js/src/shell/Makefile.in
index a7d341a..e914b6a 100644
--- a/js/src/shell/Makefile.in
+++ b/js/src/shell/Makefile.in
@@ -70,3 +70,5 @@ endif
 # People expect the js shell to wind up in the top-level JS dir.
 libs::
 	$(INSTALL) $(IFLAGS2) $(PROGRAM) $(DEPTH)
+
+GARBAGE += $(DEPTH)/$(PROGRAM)
-- 
1.7.0.rc0.69.g2ee80

