From 7120b1dbc79e38235faf303cb36f79d83a657ab1 Mon Sep 17 00:00:00 2001
From: Mike Hommey <glandium@debian.org>
Date: Thu, 23 Jul 2009 17:31:01 +0200
Subject: [PATCH 3/5] Avoid linking with system-installed libmozjs when linking js binary to it

---
 js/src/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/js/src/Makefile.in b/js/src/Makefile.in
index 94e3c82..11ad139 100644
--- a/js/src/Makefile.in
+++ b/js/src/Makefile.in
@@ -51,7 +51,7 @@ LIBRARY_NAME	= mozjs
 GRE_MODULE	= 1
 
 SIMPLE_PROGRAMS	= js$(BIN_SUFFIX)
-LIBS		= $(NSPR_LIBS) -L. $(MOZ_JS_LIBS)
+LIBS		= -L. $(MOZ_JS_LIBS) $(NSPR_LIBS)
 # Make sure the library is built before the js binary
 $(SIMPLE_PROGRAMS): $(DLL_PREFIX)$(LIBRARY_NAME)$(DLL_SUFFIX)
 
-- 
1.7.0.rc0.69.g2ee80

