From 2f147f37ddb6cb5ec966af4e1195c7b8a0e954ff Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Sat, 19 Feb 2011 11:01:36 +0100
Subject: Unbrand Browser Glue Service

---
 browser/components/nsBrowserGlue.js |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/browser/components/nsBrowserGlue.js b/browser/components/nsBrowserGlue.js
index 6624b62..45f9089 100644
--- a/browser/components/nsBrowserGlue.js
+++ b/browser/components/nsBrowserGlue.js
@@ -1087,7 +1087,7 @@ BrowserGlue.prototype = {
 
 
   // for XPCOM
-  classDescription: "Firefox Browser Glue Service",
+  classDescription: "Browser Glue Service",
   classID:          Components.ID("{eab9012e-5f74-4cbc-b2b5-a590235513cc}"),
   contractID:       "@mozilla.org/browser/browserglue;1",
 
-- 
1.7.4.1

