From: Mike Hommey <mh@glandium.org>
Date: Thu, 30 Jun 2011 14:47:36 +0200
Subject: Allow ARMv4T in nanojit

---
 js/src/nanojit/njconfig.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/js/src/nanojit/njconfig.cpp b/js/src/nanojit/njconfig.cpp
index 54ab942..19b26ef 100644
--- a/js/src/nanojit/njconfig.cpp
+++ b/js/src/nanojit/njconfig.cpp
@@ -96,7 +96,7 @@ namespace nanojit
         harden_nop_insertion = false;
 
 #if defined(NANOJIT_ARM)
-        NanoStaticAssert(NJ_COMPILER_ARM_ARCH >= 5 && NJ_COMPILER_ARM_ARCH <= 7);
+        NanoStaticAssert(NJ_COMPILER_ARM_ARCH >= 4 && NJ_COMPILER_ARM_ARCH <= 7);
         arm_arch = NJ_COMPILER_ARM_ARCH;
         arm_vfp = (arm_arch >= 7);
 
