From: Mike Hommey <glandium@debian.org>
Date: Mon, 19 Jul 2010 16:59:50 +0200
Subject: Build the test plugin on GNU/kfreebsd

---
 modules/plugin/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules/plugin/Makefile.in b/modules/plugin/Makefile.in
index 6c84714..6d4fed4 100644
--- a/modules/plugin/Makefile.in
+++ b/modules/plugin/Makefile.in
@@ -53,7 +53,7 @@ DIRS		+= base/src
 #  buildable on other non-GTK unix builds
 
 ifdef ENABLE_TESTS
-ifneq (,$(filter WINNT Darwin Linux OS2 SunOS,$(OS_ARCH)))
+ifneq (,$(filter gtk2 cocoa windows android qt os2,$(MOZ_WIDGET_TOOLKIT)))
 DIRS += test
 TOOL_DIRS += sdk
 endif
