From: Mike Hommey <glandium@debian.org>
Date: Wed, 29 Oct 2008 21:00:17 +0100
Subject: Do exec instead of uselessly forking in xulrunner launcher script

---
 build/unix/mozilla.in |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/build/unix/mozilla.in b/build/unix/mozilla.in
index c85b201..e0940fc 100644
--- a/build/unix/mozilla.in
+++ b/build/unix/mozilla.in
@@ -136,8 +136,5 @@ if [ $debugging = 1 ]
 then
   echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@"
 fi
-"$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
-exitcode=$?
-
-exit $exitcode
+exec "$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
 # EOF.
