From 8e016ea565455e72035130671e05fdb3ce202b33 Mon Sep 17 00:00:00 2001
From: Mike Hommey <glandium@debian.org>
Date: Wed, 29 Oct 2008 21:00:17 +0100
Subject: [PATCH 25/44] Do exec instead of uselessly forking in xulrunner launcher script

---
 build/unix/mozilla.in |    5 +----
 1 files changed, 1 insertions(+), 4 deletions(-)

diff --git a/build/unix/mozilla.in b/build/unix/mozilla.in
index 4f54e23..3079a12 100644
--- a/build/unix/mozilla.in
+++ b/build/unix/mozilla.in
@@ -134,8 +134,5 @@ if [ $debugging = 1 ]
 then
   echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@"
 fi
-"$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
-exitcode=$?
-
-exit $exitcode
+exec "$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
 # EOF.
-- 
1.7.0.rc0.69.g2ee80

