From eb30c94708bed9a2e36423e7bfda8aa74fce45a5 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Tue, 23 Feb 2010 13:40:34 +0100
Subject: [PATCH 28/40] Don't build the null plugin

---
 modules/plugin/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules/plugin/Makefile.in b/modules/plugin/Makefile.in
index aef5287..5da91ea 100644
--- a/modules/plugin/Makefile.in
+++ b/modules/plugin/Makefile.in
@@ -57,7 +57,7 @@ endif
 #  buildable on other non-GTK unix builds
 
 ifdef MOZ_ENABLE_GTK2
-TOOL_DIRS += default/unix
+#TOOL_DIRS += default/unix
 endif
 
 ifeq ($(OS_ARCH),WINNT)
-- 
1.7.0.rc0.69.g2ee80

