From f374521835e300254d63696533c1e08c24e8c6f4 Mon Sep 17 00:00:00 2001
From: Mike Hommey <mh@glandium.org>
Date: Wed, 27 Jan 2010 12:13:17 +0100
Subject: [PATCH 33/44] Disable python-xpcom tests for now.

---
 extensions/python/xpcom/Makefile.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/extensions/python/xpcom/Makefile.in b/extensions/python/xpcom/Makefile.in
index 1caef5f..0197530 100644
--- a/extensions/python/xpcom/Makefile.in
+++ b/extensions/python/xpcom/Makefile.in
@@ -45,9 +45,9 @@ DIRS = \
         components \
         $(NULL)
 
-ifdef ENABLE_TESTS
-        DIRS += test $(NULL)
-endif
+#ifdef ENABLE_TESTS
+#        DIRS += test $(NULL)
+#endif
 
 topsrcdir	= @top_srcdir@
 srcdir		= @srcdir@
-- 
1.7.0.rc0.69.g2ee80

