From: Mike Hommey <glandium@debian.org>
Date: Mon, 3 May 2010 10:05:48 +0200
Subject: Remove VISIBILITY_FLAGS hack in ipc/chromium/Makefile.in

---
 ipc/chromium/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/ipc/chromium/Makefile.in b/ipc/chromium/Makefile.in
index d4294bf..8fd7f3a 100644
--- a/ipc/chromium/Makefile.in
+++ b/ipc/chromium/Makefile.in
@@ -78,7 +78,7 @@ vpath %.mm \
 # instantiates template declarations in wrapped system headers with
 # the wrong visibility, which is causing the Linux tinderboxen to die in
 # std::string internals.
-VISIBILITY_FLAGS=
+#VISIBILITY_FLAGS=
 
 include $(topsrcdir)/config/config.mk
 include $(topsrcdir)/ipc/chromium/chromium-config.mk
