From 8ecb7c7d17bf48697012fc8d4af0ef7335378900 Mon Sep 17 00:00:00 2001
From: Mike Hommey <glandium@debian.org>
Date: Fri, 20 Aug 2010 18:43:18 +0200
Subject: Add a 2 minutes timeout on xpcshell tests

---
 testing/xpcshell/head.js |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/testing/xpcshell/head.js b/testing/xpcshell/head.js
index 75459eb..8f8d654 100644
--- a/testing/xpcshell/head.js
+++ b/testing/xpcshell/head.js
@@ -104,9 +104,15 @@ function _do_main() {
   var thr = Components.classes["@mozilla.org/thread-manager;1"]
                       .getService().currentThread;
 
+  var timer = Components.classes["@mozilla.org/timer;1"]
+                        .createInstance(Components.interfaces.nsITimer);
+  timer.initWithCallback(function () { _passed = false; dump("TEST-UNEXPECTED-FAIL | test is taking more than 2 minutes\n"); _do_quit() }, 120000, timer.TYPE_ONE_SHOT);
+
   while (!_quit)
     thr.processNextEvent(true);
 
+  timer.cancel();
+
   while (thr.hasPendingEvents())
     thr.processNextEvent(true);
 }
-- 
1.7.2.rc1.13.gd67bc

