From: Mike Hommey <mh@glandium.org>
Date: Fri, 18 Mar 2011 08:46:05 +0100
Subject: Allow to build against SQLite 3.7.3

---
 configure.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.in b/configure.in
index 3da41a8..c42ccce 100644
--- a/configure.in
+++ b/configure.in
@@ -130,7 +130,7 @@ GCONF_VERSION=1.2.1
 GIO_VERSION=2.0
 STARTUP_NOTIFICATION_VERSION=0.8
 DBUS_VERSION=0.60
-SQLITE_VERSION=3.7.4
+SQLITE_VERSION=3.7.3
 LIBNOTIFY_VERSION=0.4
 
 MSMANIFEST_TOOL=
