From: Mike Hommey <mh@glandium.org>
Date: Tue, 23 Feb 2010 13:40:34 +0100
Subject: Don't build the null plugin

---
 modules/plugin/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules/plugin/Makefile.in b/modules/plugin/Makefile.in
index 06ebc7b..4e9e2d6 100644
--- a/modules/plugin/Makefile.in
+++ b/modules/plugin/Makefile.in
@@ -57,7 +57,7 @@ endif
 #  buildable on other non-GTK unix builds
 
 ifdef MOZ_ENABLE_GTK2
-TOOL_DIRS += default/unix
+#TOOL_DIRS += default/unix
 endif
 
 ifeq ($(OS_ARCH),WINNT)
