From: Mike Hommey <mh@glandium.org>
Date: Fri, 17 Feb 2012 16:32:32 +0100
Subject: Bug 723861 - Ship xpidllex.py and xpidlyacc.py

---
 xpcom/typelib/xpidl/Makefile.in |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/xpcom/typelib/xpidl/Makefile.in b/xpcom/typelib/xpidl/Makefile.in
index cffceee..30f1a66 100644
--- a/xpcom/typelib/xpidl/Makefile.in
+++ b/xpcom/typelib/xpidl/Makefile.in
@@ -48,6 +48,8 @@ SDK_BINARY = \
   $(topsrcdir)/xpcom/idl-parser/xpidl.py \
   $(topsrcdir)/xpcom/idl-parser/header.py \
   $(topsrcdir)/xpcom/idl-parser/typelib.py \
+  $(DEPTH)/xpcom/idl-parser/xpidllex.py \
+  $(DEPTH)/xpcom/idl-parser/xpidlyacc.py \
   $(NULL)
 
 $(SDK_BIN_DIR)/ply::
