From: Ginn Chen <ginn.chen@oracle.com>
Date: Wed, 23 Feb 2011 15:18:17 +0800
Subject: Bug 607900 [startup-notification] Restore DESKTOP_STARTUP_ID if a restart is required r=benjamin sr=glandium a=beltzner

---
 toolkit/xre/nsAppRunner.cpp |   27 ++++++++++++++++-----------
 1 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 68e85b9..650ebbe 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -873,7 +873,7 @@ nsXULAppInfo::GetUserCanElevate(PRBool *aUserCanElevate)
     //      be elevated again)
     //   TokenElevationTypeLimited: The token is linked to a limited token 
     //     (e.g. UAC is enabled and the user is not elevated, so they can be
-    //	    elevated)
+    //      elevated)
     *aUserCanElevate = (elevationType == VistaTokenElevationTypeLimited);
   }
 
@@ -3361,6 +3361,18 @@ XRE_main(int argc, char* argv[], const nsXREAppData* aAppData)
       return 1;
     }
 
+#if defined(HAVE_DESKTOP_STARTUP_ID) && defined(MOZ_WIDGET_GTK2)
+    // DESKTOP_STARTUP_ID is cleared now,
+    // we recover it in case we need a restart.
+    if (!desktopStartupID.IsEmpty()) {
+      nsCAutoString desktopStartupEnv;
+      desktopStartupEnv.AssignLiteral("DESKTOP_STARTUP_ID=");
+      desktopStartupEnv.Append(desktopStartupID);
+      // Leak it with extreme prejudice!
+      PR_SetEnv(ToNewCString(desktopStartupEnv));
+    }
+#endif
+
 #if defined(MOZ_UPDATER) && !defined(ANDROID)
     // Check for and process any available updates
     nsCOMPtr<nsIFile> updRoot;
@@ -3702,6 +3714,9 @@ XRE_main(int argc, char* argv[], const nsXREAppData* aAppData)
           if (toolkit && !desktopStartupID.IsEmpty()) {
             toolkit->SetDesktopStartupID(desktopStartupID);
           }
+          // Clear the environment variable so it won't be inherited by
+          // child processes and confuse things.
+          g_unsetenv ("DESKTOP_STARTUP_ID");
 #endif
 
 #ifdef XP_MACOSX
@@ -3819,16 +3834,6 @@ XRE_main(int argc, char* argv[], const nsXREAppData* aAppData)
       }
 #endif
 
-#if defined(HAVE_DESKTOP_STARTUP_ID) && defined(MOZ_WIDGET_GTK2)
-      if (!desktopStartupID.IsEmpty()) {
-        nsCAutoString desktopStartupEnv;
-        desktopStartupEnv.AssignLiteral("DESKTOP_STARTUP_ID=");
-        desktopStartupEnv.Append(desktopStartupID);
-        // Leak it with extreme prejudice!
-        PR_SetEnv(ToNewCString(desktopStartupEnv));
-      }
-#endif
-
 #ifdef MOZ_WIDGET_GTK2
       MOZ_gdk_display_close(display);
 #endif
