From: Mike Hommey <mh@glandium.org>
Date: Mon, 6 Jun 2011 07:08:55 +0200
Subject: Bug 662223 - Add -lmozalloc in link flags provided in libxul.pc

---
 xulrunner/installer/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/xulrunner/installer/Makefile.in b/xulrunner/installer/Makefile.in
index dfa78a8..19a8004 100644
--- a/xulrunner/installer/Makefile.in
+++ b/xulrunner/installer/Makefile.in
@@ -102,7 +102,7 @@ FULL_NSPR_LIBS=$(subst $(prefix),\$${sdkdir},$(shell $(DEPTH)/nsprpub/config/nsp
 NSPR_VERSION=$(shell $(DEPTH)/nsprpub/config/nspr-config --version)
 endif
 
-MOZ_XUL_LINK = -lxpcomglue_s -lxul -lxpcom
+MOZ_XUL_LINK = -lxpcomglue_s -lxul -lxpcom -lmozalloc
 ifdef JS_SHARED_LIBRARY
 MOZ_JS_LINK = -lmozjs
 else
