From: Mike Hommey <mh@glandium.org>
Date: Fri, 25 Feb 2011 13:35:48 +0100
Subject: Bug 589744 - Fallback to perf measurement stub when perf_event_open syscall isn't supported

---
 js/src/configure.in |   17 ++++++++++++++++-
 1 files changed, 16 insertions(+), 1 deletions(-)

diff --git a/js/src/configure.in b/js/src/configure.in
index fdaec65..8884e22 100644
--- a/js/src/configure.in
+++ b/js/src/configure.in
@@ -3477,7 +3477,22 @@ case $target in
 esac
 
 dnl Performance measurement headers.
-AC_CHECK_HEADER(linux/perf_event.h, HAVE_LINUX_PERF_EVENT_H=1)
+AC_CHECK_HEADER(linux/perf_event.h,
+    [AC_CACHE_CHECK(for perf_event_open system call,ac_cv_perf_even_open,
+        [AC_TRY_COMPILE([#include <sys/syscall.h>],[
+#ifdef __NR_perf_event_open
+    return 0
+#else
+#error no __NR_perf_event_open
+#endif
+        ],
+        ac_cv_perf_even_open=yes,
+        ac_cv_perf_even_open=no)])])
+if test "$ac_cv_perf_even_open" = "yes"; then
+    HAVE_LINUX_PERF_EVENT_H=1
+else
+    HAVE_LINUX_PERF_EVENT_H=
+fi
 AC_SUBST(HAVE_LINUX_PERF_EVENT_H)
 
 dnl Checks for libraries.
