From: Mike Hommey <mh+mozilla@glandium.org>
Date: Wed, 13 Apr 2011 09:44:23 +0200
Subject: Bug 644692 - Fix bsdiff build failure with system bzip2. r=ted

---
 other-licenses/bsdiff/Makefile.in |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/other-licenses/bsdiff/Makefile.in b/other-licenses/bsdiff/Makefile.in
index d9132ba..5f281b0 100644
--- a/other-licenses/bsdiff/Makefile.in
+++ b/other-licenses/bsdiff/Makefile.in
@@ -48,7 +48,11 @@ include $(DEPTH)/config/autoconf.mk
 HOST_PROGRAM = mbsdiff$(BIN_SUFFIX)
 HOST_CSRCS = bsdiff.c
 
+ifdef MOZ_NATIVE_BZ2
+HOST_LIBS += $(BZ2_LIBS)
+else
 HOST_LIBS += $(DIST)/host/lib/$(LIB_PREFIX)hostbz2.$(LIB_SUFFIX)
+endif
 
 ifneq (,$(filter WINCE WINNT,$(HOST_OS_ARCH)))
 HOST_EXTRA_LIBS += $(call EXPAND_LIBNAME,Ws2_32)
