From: Mike Hommey <mh@glandium.org>
Date: Fri, 25 Feb 2011 12:55:00 +0100
Subject: Bug 620931 part 5 - Enable omni.jar by default on xulrunner

---
 xulrunner/confvars.sh |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/xulrunner/confvars.sh b/xulrunner/confvars.sh
index 937da9c..20896cb 100755
--- a/xulrunner/confvars.sh
+++ b/xulrunner/confvars.sh
@@ -41,6 +41,7 @@ MOZ_APP_DISPLAYNAME=XULRunner
 MOZ_UPDATER=1
 MOZ_XULRUNNER=1
 MOZ_ENABLE_LIBXUL=1
+MOZ_CHROME_FILE_FORMAT=omni
 MOZ_STATIC_BUILD_UNSUPPORTED=1
 MOZ_APP_VERSION=$MOZILLA_VERSION
 if test "$MOZ_STORAGE"; then
