From: Mike Hommey <mh@glandium.org>
Date: Fri, 10 Feb 2012 08:26:00 +0100
Subject: Bug 721737 - Add toolkit l10n Makefiles to browser/makefiles.sh

---
 browser/makefiles.sh |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/browser/makefiles.sh b/browser/makefiles.sh
index 7868af8..a2c4946 100644
--- a/browser/makefiles.sh
+++ b/browser/makefiles.sh
@@ -85,6 +85,12 @@ browser/themes/Makefile
 $MOZ_BRANDING_DIRECTORY/Makefile
 $MOZ_BRANDING_DIRECTORY/content/Makefile
 $MOZ_BRANDING_DIRECTORY/locales/Makefile
+toolkit/locales/Makefile
+extensions/spellcheck/locales/Makefile
+intl/locales/Makefile
+netwerk/locales/Makefile
+dom/locales/Makefile
+security/manager/locales/Makefile
 "
 
 if [ "$MOZ_SAFE_BROWSING" ]; then
